### PR TITLE
Added other components API.

### PR DESCRIPTION
**Pull request by anatoo**
@argelius Please merge this changes if there is no problem.

Summary:
 * Added Angular2 API for `ons-popover`, `ons-modal`, `ons-alert-dialog` and `ons-dialog`.
 * Renamed `PageParams` to `Params`.
 * Removed unneeded some directives.

### CODE REVIEW

File: bindings/angular2/examples/alert-dialog.ts [hunk 1]
```diff
@@ -3,73 +3,79 @@ import {
...
+  ONS_DIRECTIVES,
+  Params,
+  OnInit,
+  OnDestroy
 } from '../src/angular2-onsenui';
 
 @Component({
   selector: 'div',
-  directives: [OnsAlertDialog],
+  directives: [ONS_DIRECTIVES],
```
Comment (argelius) [inline bindings/angular2/examples/alert-dialog.ts]: I think this parameter has been deprecated in favor of `NgModule`, maybe we should update the examples to work with the latest RC version.

File: bindings/angular2/examples/modal.ts [hunk 2]
```diff
@@ -1,30 +1,75 @@
...
-export class AppComponent {
+export class AppComponent implements OnInit, OnDestroy {
+  private _modal: any;
+  private _destroyModal: Function;
+
+  constructor(private _modalFactory: ModalFactory) {
+  }
+
+  ngOnInit() {
+    this._modalFactory
```
Comment (argelius) [inline bindings/angular2/examples/modal.ts]: I like this API :)

:+1:

File: bindings/angular2/examples/alert-dialog.ts (hunk 1 above)
Comment (anatoo) [inline bindings/angular2/examples/alert-dialog.ts]: Yes, it should be NgModule. I will update NgModule related code on other pull-request.

**Comment by argelius** [comment]
Looks great. I like the services for creating modals and dialog! :+1: 

I will merge this and as I mentioned in the comments the `directives` parameter of the `@Component` decorator has been deprecated and will be removed in the next RC version.

I think we should update so we are compatible with RC5+ and ignore previous versions since they are not compatible with eachother.

**Pull request merged**